workers/unity/Packages/com.improbable.gdk.core/Utility/CommandLineUtility.cs

```diff
@@ -5,6 +5,36 @@ namespace Improbable.Gdk.Core
 {
     public static class CommandLineUtility
     {
+        public static Dictionary<string, string> GetArguments()
+        {
+#if UNITY_ANDROID
+            try
+            {
+                var unityPlayer = new UnityEngine.AndroidJavaClass("com.unity3d.player.UnityPlayer");
+                var currentActivity = unityPlayer.GetStatic<UnityEngine.AndroidJavaObject>("currentActivity");
+
+                var intent = currentActivity.Call<UnityEngine.AndroidJavaObject>("getIntent");
+                var hasExtra = intent.Call<bool>("hasExtra", "arguments");
+
+                if (hasExtra)
+                {
+                    var extras = intent.Call<UnityEngine.AndroidJavaObject>("getExtras");
+                    var arguments = extras.Call<string>("getString", "arguments");
+                    UnityEngine.Debug.Log(arguments);
+                    return ParseCommandLineArgs(arguments.Split(' '));
+                }
+            }
+            catch (Exception e)
+            {
+                UnityEngine.Debug.LogException(e);
+            }
+
+            return new Dictionary<string, string>();
+#else
```

Comment:
Ideally pull this out into a method to have the `#ifdef` around the least code possible. Otherwise it becomes hard to see at a glance what this code can run on. As the other parse logic is already pulled out into a method it would also match that.